test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java

```diff
@@ -1806,8 +1805,8 @@ protected TestCluster buildTestCluster(Scope scope, long seed) throws IOExceptio
             ArrayList<Class<? extends Plugin>> mocks = new ArrayList<>(mockPlugins);
             // add both mock plugins - local and tcp if they are not there
             // we do this in case somebody overrides getMockPlugins and misses to call super
-            if (mockPlugins.contains(MockTcpTransportPlugin.class) == false) {
-                mocks.add(MockTcpTransportPlugin.class);
+            if (mockPlugins.contains(randomTestTransportPlugin()) == false) {
+                mocks.add(randomTestTransportPlugin());
```

Comment:
Maybe we could use a different name for these two methods then. Prefixing them with `random` implies to me each call may return a different value.  Maybe just `getTestTransportPlugin()` and `getTestTransportType()`?  I don't think key is right, since the setting it is a value for is called `transport.type`?